src/movement.rs

```diff
@@ -1,293 +1,134 @@
 use bevy::{
-    ecs::system::SystemParam,
-    math::{Quat, Vec2, Vec3},
-    prelude::{
-        Commands, Component, Deref, DerefMut, Entity, EventWriter, Query, Res, ResMut, Transform,
-        With,
-    },
-    time::{Time, Timer},
+    math::{Quat, Vec2},
+    prelude::*,
+    time::Time,
 };
-use leafwing_input_manager::prelude::ActionState;
+use iyes_loopless::prelude::*;
 
 use crate::{
-    animation::Facing,
     consts::{self, LEFT_BOUNDARY_MAX_DISTANCE},
     enemy::SpawnLocationX,
-    input::PlayerAction,
     metadata::{GameMeta, LevelMeta},
-    state::State,
-    ArrivedEvent, DespawnMarker, Player, Stats,
+    GameState, Player,
 };
 
-#[cfg_attr(feature = "debug", derive(bevy_inspector_egui::Inspectable))]
-#[derive(Component, Deref, DerefMut)]
-pub struct MoveInDirection(pub Vec2);
-
-// (Moving) bondary before which, the players can't go back.
-#[derive(Component)]
-pub struct LeftMovementBoundary(f32);
-
-impl Default for LeftMovementBoundary {
-    fn default() -> Self {
-        Self(-LEFT_BOUNDARY_MAX_DISTANCE)
+/// Plugin handling movement and rotation through velocities and torques.
```

Comment:
I was pre-annoyed at the thought of having to work with torques, but after reading further seems like that really only comes into play in application of velocity systems so :+1: